### ISSUE TITLE
14_Code_9_Alpha_Awaaz.A-Eye

### ISSUE THREAD

**Issue by IshanJawade** [CodeBreakMIT/CodeBreak1.0#6]
**Team Number**: 14
**Team Name**: Code 9 Alpha
**Project Name**: Awaaz.A-Eye

## ℹ️ Project information

- **Project Name**:  Awaaz.A-Eye
- **Short Project Description**:  To write a code for a cause to make visually impaired and vocally challenged people independent and self-reliant using object detection, optical character recognition, and image classification.
- **Team ID**: 14
- **Team Name**: Code 9 Alpha
- **Team Members**: 
Ameya Kale (https://github.com/AmeyaK17)
Ishan Jawade (https://github.com/IshanJawade)
Pratik Kakade (https://github.com/pandykad)
Rushikesh Jadhav (https://github.com/Unknown-reality)

- **Demo Link**: https://drive.google.com/open?id=1EQz88wk8xoU_sMkUrFb6FHgh0xhUn1wI
- **Repository Link**: https://github.com/IshanJawade/Awaaz.A-Eye
- **Labels**: Machine Learning, Deep Learning, Tensorflow, Keras, Flutter, Python, Python3.7.+, Dart,  

## 🔥 Your Pitch
Awaaz.A-Eye is an open-source, mobile-based application solution for visually impaired and vocally challenged people. There are many smart devices available for such people like Braille eBook Reader, Smart glasses, etc .which requires hardware but is costly. Our mobile application is different and something new. There is no such application which gives the output as audio while also being a cheap alternative.

The app is developed using the Flutter Framework which makes it easier to support both iOS and Android platforms. 

The user basically gets three use cases: 
1)	Object Detection
2)	OCR
3)	Sign language recognition

The first two uses are targeted for the visually impaired people, while the last one is developed for the vocally challenged people.

**Module 1: Object Detection**
The user just has to point the mobile devices’ camera at the environment and the app starts object detection based on the YOLO TensorFlow Lite model. This model is specifically designed to perform machine learning on the device locally and without an internet connection. As a result, we get a much faster detection speed. This is highly important in an environment that is constantly changing.

**Module 2: OCR**
OCR stands for Optical Character Recognition. The task of OCR is to convert the text inside an image into an actual image that can be edited like normal text. This model can be used to read out newspapers or articles to blind people. But there are many applications out there that perform this exact task. We have specially included this model in our app to help these people with recognizing the information on a product like its price etc. We have applied various filters to only focus on the text that gives useful or important information about a product, like its price, expiry date, weight, etc. This helps us save computation power over recognizing and processing fewer data. We have used ML-Kit to achieve cloud-based OCR recognition to help achieve more accurate and precise results.

**Module 3: Sign Language Recognition**
There are often communication gaps while communicating with the vocally challenged people. Not everyone is able to learn and adapt sign languages that help these people communicate with people who have the ability of speech. To overcome this barrier, we have developed a machine learning-based solutions that take the sign language as an input and gives the output in a speech to the user. We have developed this solution for ASL (American Sign Language). The ML model has a huge data set of ASL symbols in different lighting conditions and taken at various angles. This helps in the accuracy of the model. This helps the vocally challenged people communicate effectively with others. 

Our goal is to achieve these solutions in a cost-effective way, the app should work in remote places with limited or no internet connectivity and provide precise, if not accurate results.

Hope we have helped make this world a better place! 😃

## Technology Stack Used
- Tensorflow 
- Keras
- Flutter
- Jupyter Notebook
- Pyhton3.7
- Android Studio
- JAVA

## 🔦 Any other specific thing you want to highlight?
Our motto is a cost-effective, Hardware-independent solution. Our team was focused on making the application internet free so it can be used not only in low coverage areas in urban cities but also in rural places where approx 65% of the Indian population resides!

## ✅ Checklist

**Before you post the issue**:
- [✓] You have followed the issue title format.
- [✓] You have mentioned the correct labels.
- [✓] You have provided all the information correctly.
